content/en/blog/_posts/Monitoring Kubernetes workloads: the sidecar pattern

```diff
@@ -0,0 +1,174 @@
+4. **Kube-state-metrics** for a big picture view at the Kubernetes cluster level.
+
+The dynamic nature of Kubernetes requires a dynamic approach to monitoring that can contend with high volumes of distributed applications and ephemeral infrastructure.
+
+## Kubernetes monitoring strategies
+
+Broadly speaking, here are the monitoring strategies as I see them:
+ 
+- **Remote polling** - The traditional service check approach that polls devices and reports on their health.
+- **Node-based (agent per host)** - A monitoring agent lives on the Kubernetes host or is operated as a DaemonSet inside the Kubernetes cluster with the appropriate configuration to gain access to info about system resources.
+- **Sidecars (agent per pod)** - The approach we’ll discuss below that monitors the Kubernetes system as well as the workloads it’s running. 
+- **Logs & APMs** - Management of both log data and application performance, which is out of scope for today’s discussion.
+
+With these observability approaches in mind, let’s take a closer look at one of the more traditional ways for monitoring Kubernetes: using Prometheus. 
+
+### Prometheus + Kubernetes
+
+Sean writes about [the ways that Prometheus can be a great telemetry-based monitoring companion to Kubernetes](https://blog.sensu.io/monitoring-kubernetes-docker-part-2-prometheus). In particular, if you don’t want your developers to have to think about modifying their deployment YAML, Prometheus is great for that. You can instrument your apps and expose telemetry-focused metrics. Developers can reference a ConfigMap to make use of a sidecar pattern, (more details on that below).
+
+However, if you need the additional functional service health checking, you can’t really map “Is this thing healthy or not?” to a 1 and a 0 with Prometheus. To check functional service health, you need to look elsewhere.  
+
+### A more comprehensive approach: the sidecar pattern
+
+The sidecar pattern of monitoring Kubernetes is a more dynamic approach, and the early days of its adoption have been fun to follow. In fact, sidecars as a pattern aren’t a formal convention of Kubernetes, but they’ve picked up speed as the Kubernetes community experiments and figure out what works. 
+
+Examples of Kubernetes sidecars include:
+
+- Service mesh
+- Logging platforms with agents that run as sidecars
+- Monitoring solutions like Sensu with an agent that runs as a sidecar, giving you a 1:1 pairing of a monitoring agent per collection of services.
+
+When you use the sidecar pattern, your Kubernetes pod holds the container that runs your app alongside the container that runs the Sensu agent. These containers share the same network space so your applications can talk to Sensu as if they were running in the same container or host. 
+
+![sidecar pattern for kubernetes](https://user-images.githubusercontent.com/38439765/76876119-785fe400-683f-11ea-8567-7c97dc035211.png)
```

Comment:
For the live blog post, I think this image should move into the website repo and be served from there.